src/server/api/controllers/products.controller.js

```diff
@@ -30,12 +28,18 @@ const getProducts = async (req) => {
   }
 
   try {
-    const products = knex('products')
+    const totalCount = (
+      await knex('products').count('id', { as: 'count' }).first()
+    ).count;
+    const products = await knex('products')
       .limit(limit)
       .offset(offset)
       .orderBy(sortKey, sortOrder);
```

Comment:
Nitpick: 
This is a very tiny detail but usually it's more clear to put the orderBy before pagination options (limit, offset).
Since that is the order SQL will perform actions in "get all items, sort them, and then slice the relevant page".